base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1564,23 +1565,22 @@ fn handle_possible_reorg<T: BlockchainBackend>(
             target: LOG_TARGET,
             "No reorg required, could not construct complete chain using block #{} ({}).",
             new_block.header.height,
-            new_block.hash().to_hex()
+            new_block_hash.to_hex()
         );
         return Ok(BlockAddResult::OrphanBlock);
     }
 
-    let new_block_hash = new_block.hash();
-
     // Check the accumulated difficulty of the best fork chain compared to the main chain.
-    let fork_header = find_strongest_orphan_tip(new_tips, chain_strength_comparer)?;
-    if fork_header.is_none() {
+    let fork_header = find_strongest_orphan_tip(new_tips, chain_strength_comparer)?.ok_or_else(|| {
```

Comment:
Apologies for the cosmetic changes - I couldn't help  removing the `unwrap()` call (even though it could never panic)